tests/test_lists2safebrowsing.py

```diff
@@ -97,3 +99,62 @@ def test_canonicalize(url, expected):
     documentation with some adjustments and additions.
     """
     assert canonicalize(url) == expected
+
+
+def _aux_add_domain_to_list(domain, previous_domains=set()):
+    """Auxiliary function for add_domain_to_list tests."""
+    canonicalized_domain = canonicalize(domain)
+    domain_hash = hashlib.sha256(canonicalized_domain)
+    output = []
+
+    with patch("test_lists2safebrowsing.open", mock_open()):
+        with open("test_blocklist.log", "w") as log_file:
+            added = add_domain_to_list(domain, previous_domains, set(),
+                                       log_file, output)
+            log_writes = log_file.write.call_args_list
+
+    return (added, canonicalized_domain, domain_hash, previous_domains,
+            log_writes, output)
+
+
+def test_add_domain_to_list():
+    """Test adding a domain to a blocklist."""
+    domain = "https://www.host.com"
+    (added, canonicalized_domain, domain_hash, previous_domains,
+     log_writes, output) = _aux_add_domain_to_list(domain)
+
+    expected_log_writes = [
+        call("[m] %s >> %s\n" % (domain, canonicalized_domain)),
+        call("[canonicalized] %s\n" % canonicalized_domain),
+        call("[hash] %s\n" % domain_hash.hexdigest()),
+    ]
+
+    assert added
+    assert canonicalized_domain in previous_domains
+    assert domain_hash.digest() in output
+    assert log_writes == expected_log_writes
+
```

Comment:
I looked into this a bit and the use of plain `assert` statements is often mentioned as an advantage of pytest over unittest, because it keeps the testing code simpler and more pythonic.
In addition, pytest implements [assertion introspection](https://docs.pytest.org/en/latest/assert.html#assertion-introspection-details)  in order to produce helpful failure reports.
What do you think? Is there another reason to prefer unittest's assert methods?

Reply:
No, there is no other reason to use the unittest's asset methods. Thanks for explaining this!